src/metadata.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 Veloman Yunkan <veloman.yunkan@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU  General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include "metadata.h"
+
+namespace zim
+{
+
+const Metadata::ReservedMetadataTable Metadata::reservedMetadataInfo = {
+  // name              isMandatory   minLength   maxLength   regex
+  { "Name",            true, 	      /*???*/10,  /*???*/99,  /*???*/""         },
+  { "Title",           true, 	      /*???*/10,  /*???*/30,  /*???*/""         },
+  { "Creator",         true, 	      /*???*/10,  /*???*/30,  /*???*/""         },
+  { "Publisher",       true, 	      /*???*/10,  /*???*/30,  /*???*/""         },
+  { "Date",            true, 	      10,         10,         "YYYY-MM-DD"      },
+  { "Description",     true,        10,         80,         ""                },
+  { "LongDescription", false,       10,         4000,       ""                },
```

Comment:
Thx for explaining!

Is https://wiki.openzim.org/wiki/Metadata considered the more "official" spec, as it evolves?

Reply:
> Is https://wiki.openzim.org/wiki/Metadata considered the more "official" spec, as it evolves?

Is my question going to answered, or will my comments continue to be removed?

Reply:
I don’t understand the question which seems somehow loaded…. « More official » than what? To know what is the purpose of the wiki.openzim.org pages, its enough to read them, « Metadata » article included.

Reply:
@mgaudin talks about the "spec" which is a step forward I agree.

So I'm just trying to understand what spec is being talked about.

If people are unwilling to answer the question in plain English (perhaps we have a translation problem) what I will do going forward now is email to ask him directly, as my questions are obviously no longer welcome here.

Reply:
It’s true that PR discussions are targetting developers and the ticket requester to a certain extend. As such it’s usually a bit hard to intervain if you don’t belong to one of these categories. Github issues and discussions are far more open… and slack channels are even less regulated. With emails, you really reduce the chance to get an answer at all, as we prefer to have open discussion, so it can be archived.

Reply:
> question which seems somehow loaded

> you don’t belong

Insulting and infantilizing me for asking an honest question was really not necessary.

I honestly don't know why this forum has deteriorated so dramatically (all my questions removed or hidden).

In any case I'm tired of being bullied.

So I will use private channels where I'm treated with respect.